File: src/main/java/com/ealanta/productapp/web/UrlValidator.java
package com.ealanta.productapp.web;

import java.net.URL;

/**
 * This simple validator makes sure the URL for the remote web service is valid BEFORE we attempt to use the RestTemplate
 * with an invalid URL and get a strange error message.s
 */
public class UrlValidator {

    public static boolean isValid(String url) {
        try {
            new URL(url).toURI();
            return true;
        }catch (Exception e) {
            return false;
        }
    }
}

File: src/src/main/kotlin/com/ealanta/productapp/misc/UrlValidator.kt
package com.ealanta.productapp.misc

import java.net.URL

/**
 * This simple validator makes sure the URL for the remote web service is valid BEFORE we attempt to use the RestTemplate
 * with an invalid URL and get a strange error message.s
 */
object UrlValidator {

    fun isValid(url: String): Boolean {
        try {
            URL(url).toURI()
            return true
        } catch (e: Exception) {
            return false
        }
    }
}
